bolt-driver/src/main/java/org/neo4j/ogm/drivers/bolt/request/BoltRequest.java

```diff
@@ -85,15 +92,14 @@ public BoltRequest(TransactionManager transactionManager) {
         String[] columns = null;
         for (Statement statement : query.getStatements()) {
             StatementResult result = executeRequest(statement);
-            if (columns == null) {
-                List<String> columnSet = result.keys();
-                columns = columnSet.toArray(new String[columnSet.size()]);
-            }
             RowModelResponse rowModelResponse = new RowModelResponse(result, transactionManager);
             RowModel model;
             while ((model = rowModelResponse.next()) != null) {
                 rowmodels.add(model);
             }
+            if (columns == null) {
```

Comment:
Why did this move? Won't this return always empty array because the response has been read?

Reply:
Designed to group tx management in RowModelResponse (more precisely its parent class BoltResponse), in which similar error handling is done.
Tests work fine, so I assume there are no side effects